src/Build/Evaluation/Expander.cs

```diff
@@ -3877,6 +3909,15 @@ private bool TryExecuteWellKnownFunction(out object returnVal, object objectInst
                     }
                     else if (_receiverType == typeof(IntrinsicFunctions))
                     {
+                        if (string.Equals(_methodMethodName, nameof(IntrinsicFunctions.RegisterAnalyzer), StringComparison.OrdinalIgnoreCase))
+                        {
+                            if (TryGetArg(args, out string arg0) && _loggingContext != null)
```

Comment:
nit - the second check seems much cheaper - so it should be frontloaded

```suggestion
                            if (_loggingContext != null && TryGetArg(args, out string arg0))
```